apps/settings/test/unit/panels/screen_lock_passcode/screen_lock_passcode_test.js

```diff
@@ -111,10 +114,16 @@ suite('ScreenLockPasscode > ', function() {
     });
 
     suite('passcode is different', function() {
-      setup(function() {
-        screenLockPasscode._settings.passcode = '0123';
-        screenLockPasscode._passcodeBuffer = '0000';
-        screenLockPasscode._checkPasscode();
+      setup(function(done) {
+        // set passcode to 0123
+        var promise = PasscodeHelper.set('0123');
```

Comment:
We can simply stub `PasscodeHelper.check` and then call to `screenLockPasscode._checkPasscode`.